assets/js/order-script.js

```diff
@@ -93,8 +93,8 @@ jQuery(document).ready(function($) {
 		form_inputs_data = {};
 		$form.find(':input').each( function() {
 			if (!$(this).is(':disabled')) {
-				name = $(this).attr("name");
-				name = name.split('[', 1)[0]; // for credit-note array []
+				var name = $(this).attr("name");
+				name = name.replace(/[[0-9]+]/, ''); // for credit-note array []
```

Comment:
This does not work, because instead of passing an array for the date, it will use the same name for each field and so instead of for example:
```JSON
{
	"_wcpdf_invoice_number":"1146",
	"_wcpdf_invoice_date": {
		"date":"2021-01-14",
		"hour":"10",
		"minute":"04"
	}
}
```
it is currently passing:
```
{
	"_wcpdf_invoice_number":"1146",
	"_wcpdf_invoice_date[date]":"2021-01-14",
	"_wcpdf_invoice_date[hour]":"10",
	"_wcpdf_invoice_date[minute]":"04"
}
```
Which is not parsed correctly and so the date is not updated with the regeneration data

Reply:
I think you could make it easier for yourself by doing the parsing at the PHP end, with just this on the JS end:
```JS
form_inputs_data = $form.serialize();
```
and then on the PHP end:
```PHP
parse_str($_POST['form_data'], $form_data);
```

Reply:
What you think of this?

```js
		// create an object with the form inputs data
		var form_inputs_data = {};
		var c = 0;
		$form.find(':input').each( function() {
			if (!$(this).is(':disabled')) {
				var name = $(this).attr("name");
				name     = name.replace(/[[0-9]+]/, ''); // removes the ID, eg. [198]
				var key  = name.match(/[^[\]]+(?=\])/g); // extracts the keys (date, hour, minute) to a new variable
				name     = name.split('[', 1)[0];        // removes the remaining square brackets content: [date], [hour], [minute]

				var value = $(this).val();

				if (key) {
					c++;
					if( c == 1 ) {
						form_inputs_data[name] = {};
					}
					form_inputs_data[name][key[0]] = value;
				} else {
					form_inputs_data[name] = value;
				}
			}
		} );
```

If you prefer the PHP way let me know.

Reply:
seems an unnecessary amount of trouble for something that can be done with a single line of code ;)